src/main/cpp/src/SparkResourceAdaptorJni.cpp

```diff
@@ -799,6 +805,11 @@ class spark_resource_adaptor final : public rmm::mr::device_memory_resource {
     return get_and_reset_metric(task_id, &task_metrics::time_lost_nanos);
   }
 
+  long get_and_reset_max_memory_allocated(long const task_id)
```

Comment:
Looking at the code this is combined for both CPU and GPU. I was thinking that we would have GPU and CPU memory separate. The main reasons I would like them separate is that, at least in the short term, the GPU value will be accurate, but the CPU one might not be. Also the if there is fragmentation for GPU we might see spilling when the total allocated < the size of the GPUs memory. This would give us a clue that it is happening. But the second one is kind of a long shot.

Reply:
Can you be more specific on which code path exactly is currently combined for CPU and GPU? I thought you mentioned in the previous comment that `post_alloc_success_core` should be GPU only, so I'm a bit confused.

Reply:
Line 837 of this file is calling `post_alloc_success_core` from `cpu_postalloc_success`, which is clearly not gpu for a GPU. If this is intended to be GPU only. Can we please update the name of the metric so it has gpu in it? and only update it when the allocation/free is for a GPU?

As for what I might have said I'm sorry I don't remember the context exactly, but I guess I was wrong.